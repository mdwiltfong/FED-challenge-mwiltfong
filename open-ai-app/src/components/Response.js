import React from "react";

const Response = () => {
  return `Test`;
};

export default Response;
