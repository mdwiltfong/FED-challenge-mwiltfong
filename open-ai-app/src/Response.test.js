import { render } from "@testing-library/react";
import Response from "./components/Response";

describe("Response component tests", () => {
  it("Smoke Test", () => {
    render(<Response />);
  });
  it("Snapshot Tests", () => {
    const screen = render(<Response />);
    expect(screen.asFragment()).toMatchSnapshot();
  });
  it("Renders a response that has the prompt and API response from OPEN AI", () => {
    const screen = render(
      <Response
        prompt={"write a poem about dinosaurs"}
        response={"Dinosaurs in the snow"}
      />
    );
    const prompt = screen.getByTestId("prompt");
    const response = screen.getByTestId("response");
    expect(prompt).toHaveTextContent("write a poem about dinosaurs");
    expect(response).toHaveTextContent("Dinosaurs in the snow");
  });
});
