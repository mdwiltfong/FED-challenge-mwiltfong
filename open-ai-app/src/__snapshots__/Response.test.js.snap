// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response component tests Snapshot Tests 1`] = `
<DocumentFragment>
  <li
    class="bg-light text-dark"
    style="font-weight: bold;"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-3 text-dark"
        >
          Prompt:
        </div>
        <div
          class="col text-dark"
          data-testid="prompt"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col-3 text-dark"
        >
          Response:
        </div>
        <div
          class="col text-dark"
          data-testid="response"
        />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
